mmv1/third_party/terraform/services/dataplex/resource_dataplex_aspect_type_test.go

```diff
@@ -0,0 +1,226 @@
+          {
+            "name": "name",
+            "type": "string",
+            "index": 1,
+            "constraints": {
+              "required": true
+            }
+          },
+          {
+            "name": "description",
+            "type": "string",
+            "index": 2
+          },
+          {
+            "name": "dataType",
+            "type": "string",
+            "index": 3,
+            "constraints": {
+              "required": true
+            }
+          },
+          {
+            "name": "metadataType",
+            "type": "enum",
+            "index": 4,
+            "constraints": {
+              "required": true
+            },
+            "enumValues": [
+              {
+                "name": "BOOLEAN",
+                "index": 1
+              },
+              {
+                "name": "NUMBER",
+                "index": 2
+              },
+              {
+                "name": "STRING",
+                "index": 3
+              },
+              {
+                "name": "BYTES",
+                "index": 4
+              },
+              {
+                "name": "DATETIME",
+                "index": 5
+              },
+              {
+                "name": "TIMESTAMP",
+                "index": 6
+              },
+              {
+                "name": "GEOSPATIAL",
+                "index": 7
+              },
+              {
+                "name": "STRUCT",
+                "index": 8
+              },
+              {
+                "name": "OTHER",
+                "index": 100
+              }
+            ]
+          },
+          {
+            "name": "mode",
+            "type": "enum",
+            "index": 5,
+            "enumValues": [
+              {
+                "name": "NULLABLE",
+                "index": 1
+              },
+              {
+                "name": "REPEATED",
+                "index": 2
+              },
+              {
+                "name": "REQUIRED",
+                "index": 3
+              }
+            ]
+          },
+          {
+            "name": "defaultValue",
+            "type": "string",
+            "index": 6
+          },
+          {
+            "name": "annotations",
+            "type": "map",
+            "index": 7,
+            "mapItems": {
+              "name": "label",
+              "type": "string"
+            }
+          },
+          {
+            "name": "fields",
+            "type": "array",
+            "index": 20,
+            "arrayItems": {
+              "name": "field",
+              "type": "record",
+              "typeRef": "field"
+            }
+          }
+        ]
+      }
+    }
+  ]
+}
+EOF
+}
+`, context)
+}
+
+func testAccDataplexAspectType_update(context map[string]interface{}) string {
+	return acctest.Nprintf(`
+resource "google_dataplex_aspect_type" "test_aspect_type_basic" {
+  aspect_type_id = "tf-test-aspect-type-basic%{random_suffix}"
```

Comment:
`aspect_type_id` is immutable, so I don't think it should be changed here. Effectively this means the test is recreating the resource, when we want to be testing that in-place update worked as expected.